backend/handlers/handlers.go

```diff
@@ -774,33 +786,48 @@ func (handlers *Handlers) getCoinsTotalBalance(_ *http.Request) (interface{}, er
 			return nil, err
 		}
 		amount := b.Available()
-		if _, ok := totalPerCoin[coinCode]; !ok {
-			totalPerCoin[coinCode] = amount.BigInt()
 
+		if totalBalance, exists := totalBalancesMap[coinCode]; exists {
+			totalBalance.Add(totalBalance, amount.BigInt())
+			for i := range totalsPerCoin {
+				if totalsPerCoin[i].CoinCode == coinCode {
+					totalsPerCoin[i].TotalBalance.Set(totalBalancesMap[coinCode])
+					break
+				}
+			}
```

Comment:
```suggestion
```

You can remove this for loop because the slice holds references to ` totalBalancesMap[coinCode]` which means the item in the slice will change just from this `totalBalance.Add(totalBalance, amount.BigInt())`
([example](https://go.dev/play/p/Pp1BadAvMAC))

Reply:
Makes sense, thanks.